### ISSUE TITLE
`Eq` broken

### ISSUE THREAD

**Issue by pitaj** [udoprog/fixed-map#15]
When messing around with clippy lints, I found that `#![no_std]` can be enabled unconditionally in the main crate.
However, when looking for other `no_std` stuff, I came across the following:

https://github.com/udoprog/fixed-map/blob/62ea95057058e4b11a44486fafb6aeb3f65a350f/tests/derive_tests.rs#L1

I'm not sure why it's there, but since there is no `no_std` feature, this essentially has caused those tests to be disabled.
When I removed it, `cargo test` starts failing with the following message:
```
fixed-map$ cargo test
   Compiling fixed-map v0.8.0-alpha.2 (/home/peter/Dev/fixed-map)
    Finished test [unoptimized + debuginfo] target(s) in 0.42s
     Running unittests src/lib.rs (target/debug/deps/fixed_map-9035f9f1a380e143)

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s

     Running tests/derive_tests.rs (target/debug/deps/derive_tests-56212a1e8a4a06f9)

running 4 tests
test test_clone ... ok
test test_debug ... ok
test test_fromiter ... ok
test test_eq ... FAILED

failures:

---- test_eq stdout ----
thread 'test_eq' panicked at 'assertion failed: `(left != right)`
  left: `{First: 42}`,
 right: `{First: 42, Second: 42}`', tests/derive_tests.rs:32:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
    test_eq

test result: FAILED. 3 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s

error: test failed, to rerun pass `--test derive_tests`
```

As you can see, it appears that the two sides are, in fact, not equal, but the `Eq` impl is not returning the correct response.

**Comment by udoprog** [comment]
Thanks for the report! The attribute was a mistake.

I've been working on moving and expanding the suite of doc tests to cover what is covered by `derive_tests.rs`. Everything has been moved, but not every item has doc tests (yet).

**Comment by udoprog** [comment]
Fixed in #17
